### PR TITLE
Topic recognition: 47444243 ViT for ADNI dataset

### PR DESCRIPTION
**Pull request by STommy64**
Visual Transformer to classify MRI scans in the ADNI dataset as "Alzheimer's disease" or "normal cognition".
Contains the following files:

- dataset.py: Used to generate the dataloaders for training, validating, and testing given a file path, assumes that a 'train' and 'test' folder exists
- modules.py: Contains the ViT model and all submodules
- train.py: Contains the training loop, validating loop, and testing loop
- predict.py: Gives an example on the usage of a trained ViT

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Goos
- Only the val accuracy plot is shown. No training accuracy and validation accuracy.
- Need to compare training acc with val acc to tell how bad the overfitting is


Commit messages: good

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- Did patient-level split
- Model is overfitting
- Test acc is 61.62%


General comments:
**PR is in the main branch**

- Solves the problem appropriately although performance is poor

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present, need to use own folder for report code, restore repo README -3
No Usage & Demo & Visualisation & Data usage -1
Module present
Commenting minimal -1
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no training plot -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, need to move code into own folder, restore repo README  -2
Request Description OK

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above). Also remove gitignore to allow merge

**Comment by STommy64** [comment]
[image: image.png]
Moved original readme into root and moved existing code and report readme
into new folder
[image: image.png]

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2, though the adjustment is not fully successful. Should move the student-created-document into topic-recognition/recognition/, instead of just topic-recognition/. But give granted marks for your attempt. Try to fix it again so that Shakes can merge your pull request.

**Comment by STommy64** [comment]
Hi, is the following correct?
[image: image.png]
